### PR TITLE
DCS 908 UoF display removal request flag and link on incident page

### PR DESCRIPTION
**Pull request by GurnankCheema**
Can not deploy yet as need to set up the View removal request page
Pages without and then with a removal request

Screenshot 2021-04-06 at 10 20 52

Screenshot 2021-04-06 at 10 21 01

Screenshot 2021-04-06 at 10 33 35

Screenshot 2021-04-06 at 10 33 50

### CODE REVIEW

File: server/data/incidentClient.test.ts [hunk 1]
```diff
@@ -59,9 +59,13 @@ test('getReportForReviewer', () => {
 test('getIncompleteReportsForReviewer', () => {
   const isOverdue = `(select count(*) from "v_statement" s
                       where r.id = s.report_id 
-                      and s.statement_status = $3
+                      and (s.statement_status = $3 or s.statement_status = $4)
```
Comment (GurnankCheema) [inline server/data/incidentClient.test.ts:62]: need to select where either OVERDUE or REMOVAL_REQUESTED

File: server/views/macros.njk [hunk 2]
```diff
@@ -10,3 +10,7 @@
 <span data-qa="{{props.qa}}" class="moj-badge moj-badge--red">{{props.text}}</span>
 {% endmacro %}
 
+
+{% macro removalRequestedBadge(props) %}
```
Comment (GurnankCheema) [inline server/views/macros.njk:14]: new macro

File: server/views/pages/not-completed-incidents.html [hunk 3]
```diff
@@ -30,28 +28,25 @@
           <td class="govuk-table__cell">{{report.offenderName}}</td>
           <td class="govuk-table__cell">{{report.offenderNo}}</td>
           <td class="govuk-table__cell">{{report.staffMemberName}}</td>
+          <td class="govuk-table__cell"><a href="/{{ report.id }}/view-statements" draggable="false" class="govuk-link">View incident</a></td>
+    
           <td class="govuk-table__cell">
-            <a
```
Comment (GurnankCheema) [inline server/views/pages/not-completed-incidents.html:34]: This has been moved to the /view-statements page

File: server/views/pages/reviewer/view-statements.html [hunk 4]
```diff
@@ -30,42 +31,72 @@ <h1 class="govuk-heading-xl">{{ pageTitle }}</h1>
...
           </tr>
         </thead>
 
         <tbody class="govuk-table__body">
-            {% for statement in data.statements %}
-            <tr class="govuk-table__row">
-              <td class="govuk-table__cell">{{statement.name }}</td>
-              <td class="govuk-table__cell">
-                {% if statement.isSubmitted %}
+          {% for statement in data.statements %}
```
Comment (GurnankCheema) [inline server/views/pages/reviewer/view-statements.html:41]: New OVERDUE and REMOVAL REQUEST badges. 
If both conditions are true, then only display the REMOVAL REQUEST

File: server/views/pages/not-completed-incidents.html [hunk 5]
```diff
@@ -30,28 +28,25 @@
...
+              overdueBadge({
+                text: 'OVERDUE', 
+                qa: 'removal-request' 
+              }) 
+            }}     
+            <br>
+            {% endif %}
+            {% if report.isRemovalRequested %}
               {{ 
-                overdueBadge({
```
Comment (GurnankCheema) [inline server/views/pages/not-completed-incidents.html:50]: Display both OVERDUE and REMOVAL REQUESTED if both true

File: integration-tests/integration/coordinator/add-involved-staff.spec.js [hunk 6]
```diff
@@ -66,7 +66,7 @@ context('A use of force coordinator can add involved staff', () => {
       .statements()
       .then(result =>
         expect(result).to.deep.equal([
-          { username: 'TEST_USER name', link: 'View statement', isOverdue: false, isUnverified: false },
+          { username: 'TEST_USER name', badge: '', link: 'View statement', isOverdue: false, isUnverified: false },
```
Comment (GurnankCheema) [inline integration-tests/integration/coordinator/add-involved-staff.spec.js:69]: Introduced 'badges' in integration tests to refer to the OVERDUE, REMOVAL REQUEST and EMAIL NOT VERIFIED components. This differentiates them from the href links.

File: server/views/pages/reviewer/view-statements.html [hunk 7]
```diff
@@ -64,6 +61,13 @@ <h1 class="govuk-heading-xl">{{ pageTitle }}</h1>
                   }) 
                 }}                  
               {% endif %}
+
+              {% if not statement.isVerified and (removalRequested or statement.isOverdue) %}
```
Comment (GurnankCheema) [inline server/views/pages/reviewer/view-statements.html:65]: 'Email not verified' link relocated below other links as it has longer text. Applied top margin spacing if there is more one item to display

File: server/data/incidentClient.test.ts (hunk 1 above)
Comment (andrewrlee) [reply server/data/incidentClient.test.ts:62]: this doesn't seem right? If status was removal requested then this wouldn't be overdue?

Comment (GurnankCheema) [reply server/data/incidentClient.test.ts:62]: hmm. Yes on second thoughts, it doesn't make sense. 
The overdue status is captured a few lines below already.

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**